src/material-experimental/mdc-radio/harness/radio-harness.ts

```diff
@@ -8,7 +8,78 @@
 
 import {ComponentHarness, HarnessPredicate} from '@angular/cdk-experimental/testing';
 import {coerceBooleanProperty} from '@angular/cdk/coercion';
-import {RadioButtonHarnessFilters} from './radio-harness-filters';
+import {RadioButtonHarnessFilters, RadioGroupHarnessFilters} from './radio-harness-filters';
+
+/**
+ * Harness for interacting with a standard mat-radio-group in tests.
+ * @dynamic
+ */
+export class MatRadioGroupHarness extends ComponentHarness {
+  static hostSelector = 'mat-radio-group';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a radio-group with
+   * specific attributes.
+   * @param options Options for narrowing the search:
+   *   - `id` finds a radio-group with specific id.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: RadioGroupHarnessFilters = {}): HarnessPredicate<MatRadioGroupHarness> {
+    return new HarnessPredicate(MatRadioGroupHarness)
+        .addOption('id', options.id, async (harness, id) => (await harness.getId()) === id);
+  }
+
+  private _radioButtons = this.locatorForAll(MatRadioButtonHarness);
+
+  /** Gets the name of the radio-group. */
+  async getName(): Promise<string|null> {
+    const hostName = await (await this.host()).getAttribute('name');
+
+    // It's not possible to always determine the "name" of a radio-group
+    // by reading the attribute. This is because the radio-group does not
+    // set the "name" as an element attribute if the "name" value is bound.
+    if (hostName !== null) {
+      return hostName;
+    }
+
+    // In case we couldn't determine the "name" of a radio-group by
+    // reading the "name" attribute, we try to determine the "name"
+    // of the group by going through all radio buttons.
+    let groupName: string|null = null;
+    for (let radio of await this.getRadioButtons()) {
+      const radioName = await radio.getName();
+      if (groupName === null) {
+        groupName = radioName;
+      } else if (groupName !== radioName) {
+        // It's possible that radio-button's which are part of the same
+        // group can have a inconsistent names. In that case we can't
+        // determine the group-name properly and throw an error.
+        throw Error('Radio buttons in radio-group have inconsistent names.');
+      }
+    }
+    return groupName;
+  }
+
+  /** Gets the id of the radio-group. */
+  async getId(): Promise<string|null> {
+    return (await this.host()).getAttribute('id');
+  }
+
+  /** Gets the selected radio-button in a radio-group. */
+  async getSelectedRadioButton(): Promise<MatRadioButtonHarness|null> {
```

Comment:
I think it would also be nice to have `getSelectedValue` which just returns the value of the selected radio button

Reply:
Please see what I wrote into the PR description. Unless I miss something.. this is not possible to implement.

Reply:
I see; I I think we should bind the underlying input's `value` to the value of `MatRadioButton`. I'm assuming most radios use string/number/boolean values, so that would capture the majority of values we care about. People who use more complex types wouldn't be able to query for those, but I imagine those are relatively rare. [We already do this for checkbox](https://github.com/angular/components/blob/master/src/material/checkbox/checkbox.html#L9).

Reply:
hmm, so we would always assign the radio value to the underlying input (similar to checkbox). The harness then would just read the attribute and return that value?

Though we'd need some filtering to avoid that we return `[object Object]` or a stringed function (if a function is passed as value). Ideally we'd only set the value attribute if the value is either boolean, number or string. thoughts?

Reply:
I think we should always set the value, it will just get `toString`ed, otherwise yeah

Reply:
Done. Please have another look.